---
uint8 STOPPED=0
uint8 RUNNING=1
uint8 PAUSED=2
uint8 state
